pkg/awsclient/elb_helper.go

```diff
@@ -0,0 +1,221 @@
+	i := &elb.CreateLoadBalancerListenersInput{
+		Listeners:        listeners,
+		LoadBalancerName: aws.String(elbName),
+	}
+	_, err := c.CreateLoadBalancerListeners(i)
+	return err
+}
+
+// AddLoadBalancerInstances will attach +instanceIds+ to +elbName+
+// so that they begin to receive traffic. Note that this takes an amount of
+// time to return. This is also additive (but idempotent - TODO: Validate this).
+// Note that the recommended steps:
+// 1. stop the instance,
+// 2. deregister the instance,
+// 3. start the instance,
+// 4. and then register the instance.
+func (c *awsClient) AddLoadBalancerInstances(elbName string, instanceIds []string) error {
+	instances := make([]*elb.Instance, 0)
+	for _, instance := range instanceIds {
+		instances = append(instances, &elb.Instance{InstanceId: aws.String(instance)})
+	}
+	i := &elb.RegisterInstancesWithLoadBalancerInput{
+		Instances:        instances,
+		LoadBalancerName: aws.String(elbName),
+	}
+	_, err := c.RegisterInstancesWithLoadBalancer(i)
+	return err
+}
+
+// RemoveInstancesFromLoadBalancer removes +instanceIds+ from +elbName+, eg when an Node is deleted.
+func (c *awsClient) RemoveInstancesFromLoadBalancer(elbName string, instanceIds []string) error {
+	instances := make([]*elb.Instance, 0)
+	for _, instance := range instanceIds {
+		instances = append(instances, &elb.Instance{InstanceId: aws.String(instance)})
+	}
+	i := &elb.DeregisterInstancesFromLoadBalancerInput{
+		Instances:        instances,
+		LoadBalancerName: aws.String(elbName),
+	}
+	_, err := c.DeregisterInstancesFromLoadBalancer(i)
+	return err
+}
+
+// DoesELBExist checks for the existence of an ELB by name. If there's an AWS
+// error it is returned.
+func (c *awsClient) DoesELBExist(elbName string) (bool, *AWSLoadBalancer, error) {
+	i := &elb.DescribeLoadBalancersInput{
+		LoadBalancerNames: []*string{aws.String(elbName)},
+	}
+	res, err := c.DescribeLoadBalancers(i)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case elb.ErrCodeAccessPointNotFoundException:
+				return false, &AWSLoadBalancer{}, nil
+			default:
+				return false, &AWSLoadBalancer{}, err
+			}
+		}
+	}
+	return true, &AWSLoadBalancer{ELBName: elbName, DNSName: *res.LoadBalancerDescriptions[0].DNSName, DNSZoneId: *res.LoadBalancerDescriptions[0].CanonicalHostedZoneNameID}, nil
+}
+
+// LoadBalancerInfo fields needed from NLBs
+type LoadBalancerInfo struct {
+	ExtCanonicalHostedZoneNameID string
+	ExtDNSName                   string
+	ExtLoadBalancerArn           string
+	ExtLoadBalancerName          string
+	ExtScheme                    string
+	IntCanonicalHostedZoneNameID string
+	IntDNSName                   string
+	IntLoadBalancerArn           string
+	IntLoadBalancerName          string
+	IntScheme                    string
+}
+
+// ListAllNLBs uses the DescribeLoadBalancersV2 to get back a list of all Network Load Balancers
+func (c *awsClient) ListAllNLBs() (LoadBalancerInfo, error) {
```

Comment:
I really like the idea of combining the `elb` and `elbv2` calls into one `elb_helper.go`. 👍